libs/contrib/Data/Vect/Views/Extra.idr

```diff
@@ -0,0 +1,66 @@
+||| Additional views for Vect
+module Data.Vect.Views.Extra
+
+import Control.WellFounded
+import Data.Vect
+import Data.Nat
+
+
+||| View for splitting a vector in half, non-recursively
+public export
+data Split : Vect n a -> Type where
+    SplitNil : Split []
+    SplitOne : Split [x]
+    ||| two non-empty parts
+    SplitPair : {n : Nat} -> {m : Nat} ->
+                {x, y : a} -> {xs : Vect n a} -> {ys : Vect m a} ->
+                Split (x :: xs ++ y :: ys)
+
+total
+splitHelp : {n: Nat} -> (head : a) -> (zs : Vect n a) ->
+            Nat -> Split (head :: zs)
+splitHelp head [] _ = SplitOne
+splitHelp head (z :: zs) 0 = SplitPair {xs = []} {ys = zs}
+splitHelp head (z :: zs) 1 = SplitPair {xs = []} {ys = zs}
+splitHelp head (z :: zs) (S (S k))
+  = case splitHelp head zs k of
+         SplitOne => SplitPair {xs = []} {ys = []}
+         SplitPair {xs} {ys} => SplitPair {xs = z :: xs} {ys = ys}
```

Comment:
If you're going to always have to pass `xs` and `ys` explicitly
then they should probably be explicit arguments of the `SplitPair`
constructor.

Reply:
It's not necessary to pass `ys`'s (you can see all typechecking here: https://github.com/buzden/Idris2/commit/c2037d8450e78f5c335ce737124269a714832767)

Reply:
@buzden thanks! Can you add your commit to this PR?

Reply:
... or actually I can merge your changes myself.

PS sorry for messed up white-spaces :-/